### ISSUE TITLE
Is there any Roll-back option to get back to older version if upgrade fails

### ISSUE THREAD

**Issue by SyamalaDutt** [sparkle-project/Sparkle#1274]
Does Sparkle support any roll-back option. If the user goes for update and during update if installation fails will it revert back to older version.

**Comment by kornelski** [comment]
Yes, it always keeps the old version if the upgrade fails.

**Comment by SuryaNalluri** [comment]
@kornelski  Thank you.Can you please point me to any documentation to confirm that. I want to know clearly like what it keeps backup is  it the older pkg file. So when installation of new pkg fails it will start installing the older backup pkg.

This is really very important for me to know as my application holds some running state and configurations. When it roll-backs the state can't be lost but it needs to be saved.

**Comment by SyamalaDutt** [comment]
@kornelski Thank you.Can you please point me to any documentation to confirm that. I want to know clearly like what it keeps backup is it the older pkg file. So when installation of new pkg fails it will start installing the older backup pkg.

This is really very important for me to know as my application holds some running state and configurations. When it roll-backs the state can't be lost but it needs to be saved.

**Comment by kornelski** [comment]
Oh, I had app bundle updates in mind (from zip/tarball). 

For pkg we don't do anything other than launching Apple's installer.

**Comment by kornelski** [comment]
Updates only run after application gracefully quits, so there's no running state any more.

**Comment by SyamalaDutt** [comment]
Hi can you please point to any documentation that confirms the rollback.

**Comment by kornelski** [comment]
We probably don't have anything about this documented

**Comment by SyamalaDutt** [comment]
As you are the one of the developer team, can I proceed taking your confirmation

**Comment by kornelski** [comment]
No, for pkg installer there are no guarantees at all. If it fails, it fails. There's no recovery in Sparkle for pkg updates.

We do exactly what would happen if user launched the installer. If user launched installer and it failed, that's it.

We do not keep copies of pkg installers.

We never try to install any older version.

If your pkg installer leaves the app in broken state, the app will remain broken.

If your pkg installer fails, but doesn't break the app, the app will try updating itself next time. But Sparkle doesn't remember any history, so it will just start over as if nothing ever happened.
